### PR TITLE
Implement Deploy token for group

### PR DESCRIPTION
**Pull request by Stanislav2017**
Signed-off-by: Stanislav2017 stanislavchichek@gmail.com

### Description of your changes

Added DeployToken api for group, client side and controller side. Added referencers for relation DeployToken with Group without hardcoded IDs

Fixes https://github.com/crossplane-contrib/provider-gitlab/issues/50

I have:

- [x] Read and followed Crossplane's [contribution process].
- [x] Run `make reviewable test` to ensure this PR is ready for review.

### How has this code been tested

manually and tests

[contribution process]: https://git.io/fj2m9

### CODE REVIEW

File: examples/groups/group.yaml [hunk 1]
```diff
@@ -2,12 +2,14 @@ apiVersion: groups.gitlab.crossplane.io/v1alpha1
 kind: Group
 metadata:
   name: example-group
+  annotations:
+    crossplane.io/external-name: "12591945"
```
Comment (janwillies) [inline examples/groups/group.yaml:6]: I believe this was added by accident?

File: examples/groups/group.yaml [hunk 2]
```diff
@@ -2,12 +2,14 @@ apiVersion: groups.gitlab.crossplane.io/v1alpha1
...
   name: example-group
+  annotations:
+    crossplane.io/external-name: "12591945"
 spec:
   forProvider:
     # If not set, metadata.name will be used instead.
     name: "Example Group"
-    parentIdRef:
-      name: example-parent-group
+    # parentIdRef:
```
Comment (janwillies) [inline examples/groups/group.yaml:11]: I believe this was added by accident?

File: examples/projects/project.yaml [hunk 3]
```diff
@@ -15,3 +15,4 @@ spec:
   writeConnectionSecretToRef:
     name: gitlab-project-example-project
     namespace: crossplane-system
+  importUrlSecretRef:
```
Comment (janwillies) [inline examples/projects/project.yaml:18]: I believe this was added by accident?

File: pkg/controller/groups/deploytokens/controller.go [hunk 4]
```diff
@@ -0,0 +1,199 @@
+/*
+Copyright 2020 The Crossplane Authors.
```
Comment (janwillies) [inline pkg/controller/groups/deploytokens/controller.go:2]: 2021

**Review by janwillies — CHANGES_REQUESTED** [review]
Thanks a lot @Stanislav2017! Some small nits only.

I see some changes to the `build` submodule, I assume you updated it?

**Comment by Stanislav2017** [comment]
> Thanks a lot @Stanislav2017! Some small nits only.
> 
> I see some changes to the `build` submodule, I assume you updated it?

No, I just did a rebase with master

File: pkg/controller/groups/deploytokens/controller.go [hunk 5]
```diff
@@ -0,0 +1,199 @@
...
+}
+
+func (e *external) Create(ctx context.Context, mg resource.Managed) (managed.ExternalCreation, error) {
+	cr, ok := mg.(*v1alpha1.DeployToken)
+	if !ok {
+		return managed.ExternalCreation{}, errors.New(errNotDeployToken)
+	}
+
+	dt, _, err := e.client.CreateGroupDeployToken(
+		*cr.Spec.ForProvider.GroupID,
```
Comment (Benedikt1992) [inline pkg/controller/groups/deploytokens/controller.go:135]: I get a nil pointer exception if I leave the group field empty (which is a optional field)

Comment (Benedikt1992) [reply pkg/controller/groups/deploytokens/controller.go:135]: Implemented in #53

**Comment by Benedikt1992** [comment]
PR continued in #53

**Pull request closed without merging**